hw/bsp/ble400/pkg.yml

```diff
@@ -29,52 +29,49 @@ pkg.keywords:
 
 pkg.cflags:
     # Nordic SDK files require these defines.
-    - '-DCLOCK_ENABLED=1'
-    - '-DCOMP_ENABLED=0'
-    - '-DEGU_ENABLED=0'
-    - '-DGPIOTE_ENABLED=1'
-    - '-DI2S_ENABLED=0'
-    - '-DLPCOMP_ENABLED=1'
     - '-DNRF51'
-    - '-DPDM_ENABLED=0'
-    - '-DPERIPHERAL_RESOURCE_SHARING_ENABLED=1'
-    - '-DPWM0_ENABLED=1'
-    - '-DPWM1_ENABLED=0'
-    - '-DPWM2_ENABLED=0'
-    - '-DQDEC_ENABLED=1'
-    - '-DRNG_ENABLED=1'
-    - '-DRTC0_ENABLED=0'
-    - '-DRTC1_ENABLED=0'
-    - '-DRTC2_ENABLED=0'
-    - '-DSPI0_CONFIG_MISO_PIN=23'
-    - '-DSPI0_CONFIG_MOSI_PIN=24'
-    - '-DSPI0_CONFIG_SCK_PIN=25'
-    - '-DSPI0_ENABLED=1'
-    - '-DSPI0_USE_EASY_DMA=0'
-    - '-DSPI1_ENABLED=0'
-    - '-DSPI2_ENABLED=0'
-    - '-DSPIS0_CONFIG_MISO_PIN=4'
-    - '-DSPIS0_CONFIG_MOSI_PIN=3'
-    - '-DSPIS0_CONFIG_SCK_PIN=2'
-    - '-DSPIS0_ENABLED=0'
-    - '-DSPIS1_CONFIG_MISO_PIN=28'
-    - '-DSPIS1_CONFIG_MOSI_PIN=25'
-    - '-DSPIS1_CONFIG_SCK_PIN=29'
-    - '-DSPIS1_ENABLED=0'
-    - '-DSPIS2_ENABLED=0'
-    - '-DTIMER0_ENABLED=1'
-    - '-DTIMER1_ENABLED=0'
-    - '-DTIMER2_ENABLED=0'
-    - '-DTIMER3_ENABLED=0'
-    - '-DTIMER4_ENABLED=0'
-    - '-DTWI0_CONFIG_SCL=1'
-    - '-DTWI0_CONFIG_SDA=0'
-    - '-DTWI0_ENABLED=1'
-    - '-DTWI1_ENABLED=0'
-    - '-DTWIS0_ENABLED=0'
-    - '-DTWIS1_ENABLED=0'
-    - '-DUART0_ENABLED=1'
-    - '-DWDT_ENABLED=1'
+    - '-DNRFX_CLOCK_ENABLED=0'
+    - '-DNRFX_GPIOTE_ENABLED=0'
+    - '-DNRFX_LPCOMP_ENABLED=0'
+    - '-DNRFX_POWER_ENABLED=0'
+    - '-DNRFX_PPI_ENABLED=0'
+    - '-DNRFX_PRS_ENABLED=0'
+    - '-DNRFX_QDEC_ENABLED=0'
+    - '-DNRFX_RNG_ENABLED=0'
+    - '-DNRFX_RTC_ENABLED=0'
+    - '-DNRFX_RTC0_ENABLED=0'
+    - '-DNRFX_RTC1_ENABLED=0'
+    - '-DNRFX_SPIS_ENABLED=0'
+    - '-DNRFX_SPIS1_ENABLED=0'
+    - '-DNRFX_SPI_ENABLED=0'
+    - '-DNRFX_SPI0_ENABLED=0'
+    - '-DNRFX_SPI1_ENABLED=0'
+    - '-DNRFX_SWI_ENABLED=0'
+    - '-DNRFX_SWI0_DISABLED=0'
+    - '-DNRFX_SWI1_DISABLED=0'
+    - '-DNRFX_SWI2_DISABLED=0'
+    - '-DNRFX_SWI3_DISABLED=0'
+    - '-DNRFX_SWI4_DISABLED=0'
+    - '-DNRFX_SWI5_DISABLED=0'
+    - '-DNRFX_TIMER_ENABLED=0'
+    - '-DNRFX_TIMER0_ENABLED=0'
+    - '-DNRFX_TIMER1_ENABLED=0'
+    - '-DNRFX_TIMER2_ENABLED=0'
+    - '-DNRFX_TWI_ENABLED=0'
+    - '-DNRFX_TWI0_ENABLED=0'
+    - '-DNRFX_TWI1_ENABLED=0'
+    - '-DNRFX_UART_ENABLED=0'
+    - '-DNRFX_UART0_ENABLED=0'
+    - '-DNRFX_WDT_ENABLED=0'
+
+#todo this needs to be in the adc driver file instead, but that doesnt seem to work currently
+pkg.cflags.ADC_0:
```

Comment:
Still looking for newt feedback on these. They should be in the adc driver IMPL as theyre very related to our (wrapped abstraction) style and someone who writes a hal style doesnt want these enabled or they might be stomped on by the nordic files.